src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -508,7 +526,16 @@ def generate_config(
         }
 
         if security_enabled:
+            r2: Dict[str, Any] = {'files': {}}
+            unique_id_counter = 1
+            for url, credentials in clusters_credentials.items():
+                unique_id = unique_id_counter
+                unique_id_counter += 1
+                r2['files'][f'prometheus_{unique_id}_cert.crt'] = credentials['certificate']
+                credentials['cert_file_name'] = f'prometheus_{unique_id}_cert.crt'
+            context['clusters_credentials'] = clusters_credentials
             cert, key = self.get_mgr_prometheus_certificates(daemon_spec)
+            self.mgr.set_prometheus_cert(cert)
```

Comment:
having mTLS we will need some special certificate for the remote cluster instead of fetching/using the prometheus crt/key